src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -1291,19 +1276,14 @@ pub fn add_ssts_to_sub_level(
 }
 
 /// `None` value of `sub_level_insert_hint` means append.
-pub fn insert_new_sub_level(
+pub fn append_new_sub_level(
     l0: &mut OverlappingLevel,
     insert_sub_level_id: u64,
     level_type: PbLevelType,
     insert_table_infos: Vec<SstableInfo>,
-    sub_level_insert_hint: Option<usize>,
 ) {
-    if insert_sub_level_id == u64::MAX {
-        return;
-    }
-    let insert_pos = if let Some(insert_pos) = sub_level_insert_hint {
-        insert_pos
-    } else {
+    assert_ne!(0, insert_sub_level_id);
+    let insert_pos = {
         if let Some(newest_level) = l0.sub_levels.last() {
             assert!(
```

Comment:
The correctness of assertion depends on additional assumption of insert_sub_level_id.